.circleci/config.yml

```diff
@@ -12,12 +12,24 @@ workflows:
         - test-2.071-prod
         - test-2.077-upstream-devel
         - test-2.077-upstream-prod
+        - d2tag:
+            requires:
+            - test-1.081-devel
+            - test-1.081-prod
+            - test-2.070-devel
+            - test-2.070-prod
+            - test-2.071-devel
+            - test-2.071-prod
+            - test-2.077-upstream-devel
+            - test-2.077-upstream-prod
```

Comment:
ouch, this is really inconvenient, there is a lot of repetition in here already with the compiler versions :unamused:

Reply:
I was trying to use yaml anchors for this too but looks like there is not way to append elements to anchored array, only to an object. So you can do this:

```yaml
list: &list
- one
- two

array: *list
```

but not this:

```yaml
list: &list
- one
- two

array:
<<: *list
- three
```

This really sucks but seems to be inherent problem with YAML based configuration (i.e. we will have the same problem in Travis too if it ever gets to the point of specifying step dependencies). Can't thing of any workaround other than generating config file by script.

This is one case where Jenkins Groovy-based pipeline approach really shines as superior.

Reply:
> i.e. we will have the same problem in Travis too

Not really, not in this case, as in Travis we group all the jobs in one step and we declare dependencies (well, order really, not dependencies) between stages, not individual jobs. Maybe you can do something like that?

Reply:
https://discuss.circleci.com/t/better-way-to-define-workflow-stages/19376 though I don't have high hopes (nothing in docs hints about it).

Reply:
This really sucks. It's a lot of boilerplate and very easy to miss some place when updating.

Reply:
It will also scale terribly if we ever want to add more jobs to second stage - you will need to list all requirements per _each_ second stage job!

Reply:
:man_facepalming:

Reply:
Got some suggestions from CircleCI staff, will try the tomorrow

Reply:
No luck to avoid the repetition?

Reply:
Nope. I can use yaml anchors to paste whole array of dependencies but it will only make difference when we have more than one job at stage 2.